src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -190,6 +190,22 @@ typedef struct WorkerShardStatistics
 	HTAB *statistics;
 } WorkerShardStatistics;
 
+/* ShardMoveDependencyHashKey contains colocationId to be used as a key in a hash */
+typedef struct ShardMoveDependencyHashKey
+{
+	int64 colocationId;
+} ShardMoveDependencyHashKey;
+
+/* ShardMoveDependencyHashEntry contains the taskId which any new shard move task within the corresponding colocation group must take a dependency on */
+typedef struct ShardMoveDependencyInfo
+{
+	ShardMoveDependencyHashKey key;
+	int64 taskId;
+} ShardMoveDependencyInfo;
```

Comment:
The split of these tasks seems unnecessary now. You could store the colocationId directly in `ShardMoveDependencyInfo`. Now it's actually a bit confusing, because for nodes you store the nodeid in the colocationId. The comment also needs to be updated.